### PR TITLE
Make simplification of polygons more sophisticated

### PR DESCRIPTION
**Pull request by quis**
Simplifying polygons means reducing the number of points used to render them. This pull request implements simplification such that, for any given input polygons, the combined point count of the simplified polygons is less than 200.

# Rules

When simplifying the polygons we are trying to get the smallest number of points while meeting these two rules: 
1. No part of the area the user has chosen can be cut off
2. The area of the simplified polygon should be as small as possible

# Techniques

This pull request introduces two techniques we weren’t using before:
1. Dilating and eroding the area to fill in concave details of the shape, like inlets and harbours
2. Making the simplification threshold proportionate to the perimeter of all polygons, so bigger and crinklier polygons get more simplification applied

It also shows the estimated bleed as a separate polygon. This lets us make it bigger (so it’s more closer the the approximate bleed) without having to send a bigger area to the CBC and compounding the amount of actual bleed.

# Examples

Before | After 
---|---
image | image 
image | image
image | image
image | image
image | image
image | image
image | image

### CODE REVIEW

File: app/models/broadcast_message.py [hunk 1]
```diff
@@ -72,30 +72,24 @@ def initial_area_names(self):
...
-        return broadcast_area_libraries.get_polygons_for_areas_lat_long(
-            *self._dict['areas']
+        return Polygons(
+            list(itertools.chain(*(
+                area.polygons for area in self.areas
+            )))
         )
 
-    @property
+    @cached_property
```
Comment (servingUpAces) [inline app/models/broadcast_message.py:83]: Why is this a cached_property?

File: app/broadcast_areas/create-broadcast-areas-db.py [hunk 2]
```diff
@@ -173,3 +165,16 @@ def simplify_geometry(feature):
...
 
 repo.insert_broadcast_areas(areas_to_add)
+
+
+most_detailed_polygons = formatted_list(
+    sorted(point_counts, reverse=True)[:5],
+    before_each='',
+    after_each='',
+)
+print(  # noqa: T001
```
Comment (servingUpAces) [inline app/broadcast_areas/create-broadcast-areas-db.py:175]: Do we want to keep all the print statements?

File: app/models/broadcast_message.py (hunk 1 above)
Comment (quis) [reply app/models/broadcast_message.py:83]: We call it twice:
- once to show the actual broadcast area
- once to show the estimated bleed of the broadcast (we take the polygons for the broadcast area and buffer then by 1.5km)

Making it a cached property means we don’t need to get the JSON from the SQLite database twice, or re-smooth the polygons twice.

`cached_property` stores the result of calling the property against the object, so it’s only cached for the lifetime of that object (i.e. not between requests or anything like that).

File: app/broadcast_areas/create-broadcast-areas-db.py (hunk 2 above)
Comment (quis) [reply app/broadcast_areas/create-broadcast-areas-db.py:175]: This is only run locally, so probably doesn’t hurt? Especially when we’ll adding new areas in https://www.pivotaltracker.com/story/show/174376516

**Review by servingUpAces — APPROVED** [review]

**Pull request merged**